server/src/test/java/org/elasticsearch/action/bulk/BulkProcessorTests.java

```diff
@@ -90,6 +105,174 @@ public void testBulkProcessorFlushPreservesContext() throws InterruptedException
         bulkProcessor.close();
     }
 
+    public void testConcurrentExecutions() throws Exception {
+        final AtomicBoolean called = new AtomicBoolean(false);
+        int estimatedTimeForTest = Integer.MAX_VALUE;
+        final int simulateWorkTimeInMillis = 5;
+        int concurrentClients= 0;
+        int concurrentBulkRequests = 0;
+        int expectedExecutions= 0;
+        int maxBatchSize = 0;
+        int maxDocuments = 0;
+        int iterations = 0;
+        //find some randoms that allow this test to take under ~ 10 seconds
+        while (estimatedTimeForTest > 10_000) {
+            if (iterations++ > 1_000) {
+                fail("failed to find random values that allows test to run quickly"); //extremely unlikely
```

Comment:
instead of failing the test, use an assumeTrue/assumeFalse?

Reply:
good suggestion! done - https://github.com/elastic/elasticsearch/pull/41451/commits/a1e456c75a97ee28ff79695afc96c8503d34bf63